easybuild/tools/repository/filerepo.py

```diff
@@ -76,25 +80,33 @@ def add_easyconfig(self, cfg, name, version, stats, previous):
         # create directory for eb file
         full_path = os.path.join(self.wc, self.subdir, name)
         mkdir(full_path, parents=True)
+        yeb_format = is_yeb_format(cfg, None)
+
+        if yeb_format:
+            extension = YEB_FORMAT_EXTENSION
+            prefix = "buildstats: ["
+
+        else:
+            extension = EB_FORMAT_EXTENSION
+            prefix = "buildstats = ["
 
         # destination
-        dest = os.path.join(full_path, "%s-%s.eb" % (name, version))
+        dest = os.path.join(full_path, "%s-%s%s" % (name, version, extension))
 
         txt = "# Built with EasyBuild version %s on %s\n" % (VERBOSE_VERSION, time.strftime("%Y-%m-%d_%H-%M-%S"))
 
         # copy file
         txt += read_file(cfg)
 
         # append a line to the eb file so that we don't have git merge conflicts
-        if not previous:
-            statsprefix = "\n# Build statistics\nbuildstats = ["
-            statssuffix = "]\n"
-        else:
-            # statstemplate = "\nbuildstats.append(%s)\n"
-            statsprefix = "\nbuildstats.append("
-            statssuffix = ")\n"
-
-        txt += statsprefix + stats_to_str(stats) + statssuffix
+        statscomment = "\n# Build statistics\n"
+        statsprefix = prefix
+        statssuffix = "]\n"
+        if previous:
+            # statstemplate v= "\nbuildstats.append(%s)\n"
+            stats.update(previous)
```

Comment:
the idea is that you *add* the stats, so you have both the old & the new stats

That's not what you're doing here, it seems?